azavea/fb-gender-survey-dashboard
# Style pass

## Overview
Styles all pages to match flat designs where relevant. I didn't touch the chart colors, but can if desired. 

I also took care of some of the feedback from the client. Namely 030 (switched Answer/Question in lists), 035 (copy), 036 (copy), 039 (countries first), 040 (favicon and meta image).

~Connects #XXX~

### Demo
#### Home
<img width="1792" alt="Screen Shot 2021-01-22 at 6 04 37 PM" src="https://user-images.githubusercontent.com/5672295/105501114-3fa15e80-5cdd-11eb-9027-e884cf94a788.png">
<img width="1792" alt="Screen Shot 2021-01-22 at 6 04 44 PM" src="https://user-images.githubusercontent.com/5672295/105501125-44fea900-5cdd-11eb-98e9-61f72899dcce.png">

### Question selector
<img width="1792" alt="Screen Shot 2021-01-22 at 6 04 52 PM" src="https://user-images.githubusercontent.com/5672295/105501443-a9ba0380-5cdd-11eb-97bb-3dee8885c789.png">
<img width="1792" alt="Screen Shot 2021-01-22 at 6 04 57 PM" src="https://user-images.githubusercontent.com/5672295/105501450-aaeb3080-5cdd-11eb-8800-fe5f47a53eff.png">

<img width="1792" alt="Screen Shot 2021-01-22 at 6 04 48 PM" src="https://user-images.githubusercontent.com/5672295/105501422-a4f54f80-5cdd-11eb-9e3d-29794ca8b678.png">


#### Saved visualizations
<img width="1792" alt="Screen Shot 2021-01-22 at 6 04 24 PM" src="https://user-images.githubusercontent.com/5672295/105501208-5f388700-5cdd-11eb-92e2-216df68b6717.png">
<img width="1792" alt="Screen Shot 2021-01-22 at 6 04 10 PM" src="https://user-images.githubusercontent.com/5672295/105501276-6fe8fd00-5cdd-11eb-8b9e-7fb6aa6449d3.png">
<img width="1792" alt="Screen Shot 2021-01-22 at 6 04 20 PM" src="https://user-images.githubusercontent.com/5672295/105501287-74151a80-5cdd-11eb-9aa2-f4874f46cdb3.png">

#### Saved visualizations, empty
<img width="1791" alt="Screen Shot 2021-01-22 at 6 19 30 PM" src="https://user-images.githubusercontent.com/5672295/105502032-6613c980-5cde-11eb-9f13-f3e56803471c.png">


#### Modal styles
<img width="1792" alt="Screen Shot 2021-01-22 at 5 42 57 PM" src="https://user-images.githubusercontent.com/5672295/105501351-8727ea80-5cdd-11eb-8f09-bfbeb8c65dd0.png">
<img width="1792" alt="Screen Shot 2021-01-22 at 5 43 00 PM" src="https://user-images.githubusercontent.com/5672295/105501365-8bec9e80-5cdd-11eb-8917-d2fadf635a11.png">

#### Charts page, default
<img width="1791" alt="Screen Shot 2021-01-22 at 6 18 00 PM" src="https://user-images.githubusercontent.com/5672295/105501864-2fd64a00-5cde-11eb-96c7-14b177ba542f.png">


#### Charts page (after this page has been saved)
<img width="1792" alt="Screen Shot 2021-01-22 at 5 42 36 PM" src="https://user-images.githubusercontent.com/5672295/105501790-1503d580-5cde-11eb-97eb-d53b7b15f3fb.png">

#### 404 page
<img width="1791" alt="Screen Shot 2021-01-22 at 6 20 17 PM" src="https://user-images.githubusercontent.com/5672295/105502101-7f1c7a80-5cde-11eb-82cc-4d5f11099a6e.png">



### Notes
- This isn't fully responsive yet. I got part of the way there, but will need to do another responsive pass.
- I put in the image that was supplied by the client. I imagine we might get a new one. I placed it in a `/images/` folder that I created in `/src/` - not sure if that was right! 
- I added some text under each question that indicates the # of q's. It's hard-coded as "questions" and right now if you run a search that returns 1 item it looks awkward. I'm hoping that #73 will resolve this with the enhancement that opens all of the accordions by default.
- I would have liked to make the 404 page more consistent with the empty Saved viz page, but didn't have time to change the link to a button.

## Testing Instructions
- `git pull`
- Visit each page of the app:
    - Home
        - Note that "Countries" is the default selected type of areas. 
        - Confirm that text in Header matches the feedback in the feedback doc.
        - Open up FAQ and About. Note new styling.
        - Change browser width to see that image disappears on small screens (the rest is a bit broken but this is intentional)
    - Questions
        - Run a search that returns nothing and note that inside each question is the text that states there are no results. The checkbox should be disabled.
        - Clear your search. Open the accordions to see taht the # of questions text disappears when the accordion is open.
        - Proceed with questions from at least 2 sections.
    - Charts page
        - Each section is contained in an accordion. They are all open by default, try closing one.
        - When hovering on a chart, the download button should appear at the top right.
        - Save this selection of charts. Note the saved button is replaced by a "Saved" notice.
    - Saved Charts
        - Note your saved chart. You should always see the option to delete it, but if you hover you'll see the edit button. Click it.
        - Start editing the text. Note the new "Cancel" x button next to "Save". Click it. Your new title should not be saved.
        - Delete this chart. 
        - Note the empty state. 
    - Go to http://localhost:3333/chartssd. 
        - Note the 404 page styles.
